From e38163d9544a7dbdbe54794cd11a1697eea24ca4 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 12 Jun 2023 13:32:43 +0200
Subject: libstdc++: explicit cast of different type SSE2 intrinsics in
 CSSParser fast paths.

C++ declaration of Intel intrinsics do not allow some implicit type
conversions. That breaks calls to _mm_movemask_epi8.

Bug: 957519
Change-Id: Ie74a67df5521e9495a44f676ac562f3c07043519
---
 .../blink/renderer/core/css/parser/css_parser_fast_paths.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/core/css/parser/css_parser_fast_paths.cc b/third_party/blink/renderer/core/css/parser/css_parser_fast_paths.cc
index 96eb0f66187f0..ba8ea7b8cf8d7 100644
--- a/third_party/blink/renderer/core/css/parser/css_parser_fast_paths.cc
+++ b/third_party/blink/renderer/core/css/parser/css_parser_fast_paths.cc
@@ -284,8 +284,10 @@ static unsigned FindLengthOfValidDouble(const LChar* string, const LChar* end) {
     auto is_decimal_mask = (b >= '0' && b <= '9');
     auto is_mark_mask = (b == '.');
 #ifdef __SSE2__
-    uint16_t is_decimal_bits = _mm_movemask_epi8(is_decimal_mask);
-    uint16_t is_mark_bits = _mm_movemask_epi8(is_mark_mask);
+    uint16_t is_decimal_bits =
+        _mm_movemask_epi8(reinterpret_cast<__m128i>(is_decimal_mask));
+    uint16_t is_mark_bits =
+        _mm_movemask_epi8(reinterpret_cast<__m128i>(is_mark_mask));
 
     // Only count the first decimal mark.
     is_mark_bits &= -is_mark_bits;
-- 
2.39.2

